libs/sync/path.go

```diff
@@ -24,10 +24,13 @@ func repoPathForPath(me *iam.User, remotePath string) string {
 
 // EnsureRemotePathIsUsable checks if the specified path is nested under
 // expected base paths and if it is a directory or repository.
-func EnsureRemotePathIsUsable(ctx context.Context, wsc *databricks.WorkspaceClient, remotePath string) error {
-	me, err := wsc.CurrentUser.Me(ctx)
-	if err != nil {
-		return err
+func EnsureRemotePathIsUsable(ctx context.Context, wsc *databricks.WorkspaceClient, remotePath string, me *iam.User) error {
+	var err error
+	if me == nil {
+		me, err = wsc.CurrentUser.Me(ctx)
+		if err != nil {
+			return err
+		}
```

Comment:
Is this really needed? The additional nillable arg is a smell.

Reply:
I'd like to keep it unless we can do something better here :( This API is extremely slow and makes deployments noticeably slower. I wasn't sure at what other level in the code to add some caching for `CurrentUser.Me`. Ideally, that would be done in the SDK itself.

Reply:
Maybe we can track adding caching to the SDK as a ticket?